Update simulator state documentation

gemarcano opened the pull request:
- Added function-level documentation to the simulator core state.h
   header.

I'll add questions when I create the request.

gemarcano committed:
Update simulator state documentation

 - Added function-level documentation to the simulator core state.h
   header.

ppannuto commented on simulator/core/state.h:
```diff
@@ -19,19 +19,74 @@
 void state_pipeline_flush(uint32_t new_pc);
 #endif
 
+/** FIXME This isn't implemented anywhere?
+ *
+ */
 void state_async_block_start(void);
+
+/** FIXME This isn't implemented anywhere?
+ *
+ */
 void state_async_block_end(void);
```
Mhm, yeah. This is a "not implemented yet" idea if memory serves. There was a notion at some point that asynchronous peripherals should also be replayable (and they mostly are and it mostly works). But what happens when some peripheral state needs an atomic update (e.g. the uart peripheral head and tail pointers from its internal buffer) -- it's _possible_ that those writes get attributed to two different CPU cycles and in a pathological case that a rewind could only undo one of those writes.

This was a thought that a peripheral should be able to identify an atomic block that all gets saved to one cycle or another, but that never got implemented.

ppannuto commented on simulator/core/state.h:
```diff
@@ -19,19 +19,74 @@
+ */
 void state_async_block_end(void);
 
+/** Set the emulator's debug flag to true.
+ */
 void state_enter_debugging(void);
+
+/** Set the emulator's debug flag to false.
+ */
 void state_exit_debugging(void);
```
It may be worth noting that for state tracking, debugging is slightly more significant. In particular, it allows writes to proceed to directly affect the underlying hardware representation, rather than going in the tick/tock event queue.

ppannuto reviewed (approved):
This all looks good; some clarifications that are hopefully helpful

gemarcano committed:
simulator/core/state.h: improve documentation

 - Clarify unimplemented state_async_block_start and end functions
 - Clarify the purpose of state_start_debugging and exit

gemarcano commented:
I've rebased against latest master, and made some updates to the comments on those two sets of functions.

ppannuto reviewed (approved):
(!) This github notification got lost somewhere along the way, sorry for not seeing / merging earlier :/
